File: test/numword.py
#!/usr/bin/env python

import os, sys
dirname, filename = os.path.split(os.path.abspath(sys.argv[0]))
util_dir = dirname+"/../util/"
data_dir = dirname+"/../data/"
sys.path.append(util_dir)

from util import nonZeroToWord
import argparse

def main():
    parser = argparse.ArgumentParser(description='converts number to word')
    parser.add_argument('i', type=int, help='give a positive integer')
    args = parser.parse_args()

    i=args.i

    if (i==0):
        exit(1)
    if (i<0):
        exit(2)

    print nonZeroToWord(i).replace(" ","").replace("-","")

if __name__ == "__main__":
    main()

File: test/numword.rb
#!/usr/bin/env ruby

UTIL="util"
UTIL_DIR="../#{UTIL}"

require File.join(File.dirname(__FILE__), "#{UTIL_DIR}", "#{UTIL}")
include Util

def main
    exit 1 if ARGV.length!=1

    i=ARGV[0].to_i
    
    exit 2 if i==0
    exit 3 if i<0

    puts Util::nonZeroToWord(i,nil).gsub(' ','').gsub('-','')
end

if __FILE__==$0
    main()
end